engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1203,17 +1209,29 @@ private AllocateResponse sendContainerAskToRM(List<ContainerRequest> containerRe
       amRmClient.releaseAssignedContainer(containerId);
     }
 
+    return amRmClient.allocate(0);
+  }
+
+  private void processContainerStopRequests()
+  {
     for (String containerIdStr : dnmgr.containerStopRequests.values()) {
```

Comment:
Why can't the first for section remain in sendContainerAskToRM as that method is dealing with sending of all requests to RM.